src/sentry/discover/endpoints/discover_query.py

```diff
@@ -100,18 +100,11 @@ def post(self, request, organization):
         if not features.has("organizations:discover", organization, actor=request.user):
             return Response(status=404)
 
-        requested_projects = request.data["projects"]
-
-        projects = list(
-            Project.objects.filter(
-                id__in=requested_projects, organization=organization, status=ProjectStatus.VISIBLE
-            )
-        )
-
-        has_invalid_projects = len(projects) < len(requested_projects)
-
-        if has_invalid_projects or not request.access.has_projects_access(projects):
-            return Response("Invalid projects", status=403)
+        try:
+            requested_projects = set(map(int, request.data["projects"]))
+        except ValueError:
+            raise ResourceDoesNotExist()
+        projects = self._get_projects_by_id(requested_projects, request, organization)
```

Comment:
This changes the endpoint to return only project that the user is a member of, which I think is breaking some tests?

Reply:
I'm an idiot and forgot to pass a parameter 🤦‍♂